modules/cart/src/Form/AddToCartForm.php

```diff
@@ -179,7 +179,8 @@ public function buildEntity(array $form, FormStateInterface $form_state) {
     $entity = parent::buildEntity($form, $form_state);
     // Now that the purchased entity is set, populate the title and price.
     $entity->setTitle($entity->getPurchasedEntity()->getLineItemTitle());
-    $entity->setUnitPrice($this->chainPriceResolver->resolve($entity->getPurchasedEntity()));
+    $price = $this->chainPriceResolver->resolve($entity->getPurchasedEntity());
```

Comment:
Is resolving the price at this point necessary? I thought this was the equivalent of creating a line item, in which case we'd be able to simply do $entity->setUnitPrice($entity->getPurchasedEntity()->getPrice());